mlir/lib/Quantum/Transforms/IonsDecompositionPatterns.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright 2024 Xanadu Quantum Technologies Inc.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+//     http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#define DEBUG_TYPE "ions-decomposition"
+
+#include "Quantum/IR/QuantumOps.h"
+#include "Quantum/Transforms/Patterns.h"
+#include "VerifyParentGateAnalysis.hpp"
+#include "mlir/Dialect/Arith/IR/Arith.h"
+#include "llvm/ADT/StringSet.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Support/Errc.h"
+
+using llvm::dbgs;
+using namespace mlir;
+using namespace catalyst::quantum;
+
+constexpr double PI = 3.14159265358979323846;
```

Comment:
This caught my eye, and I just learned that C++20 (at last!) defines pi and other mathematical constants: https://en.cppreference.com/w/cpp/numeric/constants. For example:

```cpp
#include <numbers>
std::cout << std::numbers::pi << std::endl;
```

Reply:
It might require bumping to cpp20 in catalyst as a whole first

Reply:
Thank you I did not know about that, it would be nice to have. Not sure about what version we fully support right now, but  I will give it a try

Reply:
Cool, thanks! Probably not high priority but definitely nice to have :slightly_smiling_face:

Reply:
Yes I think we use `set(CMAKE_CXX_STANDARD 17 CACHE STRING "C++ standard to conform to")` in our Cmakelists

Reply:
I see @paul0403 already has a PR! #1229

Reply:
I have used `llvm::numbers::pi`